.github/workflows/ci.yaml

```diff
@@ -21,6 +21,7 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python }}
+          allow-prereleases: true
```

Comment:
I had to allow pre-releases in order to be able to fetch python 3.12 version, in setup-python action, which are not release as stable yet.